server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -414,13 +411,13 @@ private void emitMessage(KafkaProducer<String, String> producer, String topic, S
      * @param result message to write
      * @return Path to a newly written JSON file
      */
-    private String writeToFile(SingleRecord result) throws IOException, NullPointerException {
+    private String writeToFile(SingleRecord result) throws NullPointerException, IOException {
```

Comment:
Do we still need to throw IOException in the method signature?
Also, I think throwing NPE in method signature here is a bit weird but I think your intention here was not fixing this so that's fine.

Reply:
Yes, `FileWriter` throws `IOException`. It is also caught in the `handle producing` method and wrapped in `RuntimeException` to stop the plug-in whenever `IOException` happens in the FASTEN server.